aggregator.py

```diff
@@ -644,8 +660,13 @@ def submit_metric(self, name, value, mtype, tags=None, hostname=None,
 
             if context not in metric_by_context:
                 metric_class = self.metric_type_to_class[mtype]
-                metric_by_context[context] = metric_class(self.formatter, name, tags,
-                    hostname, device_name)
+                if metric_class == Histogram:
```

Comment:
This makes me sad :(
Maybe we should pass the config to the Metric objects instead ?